libexec/slurm/monitor.sh

```diff
@@ -49,21 +49,40 @@ read -r -d '' template <<'TEMPLATE' || true
 TEMPLATE
 
 # Fetch the state of the job
-control=$(scontrol show job "$1" --oneline | head -n 1 | tr ' ' '\n' 2>&1)
+raw_control=$(scontrol show job "$1" --oneline 2>&1)
 exit_status="$?"
+control=$(echo "$raw_control" | head -n 1 | tr ' ' '\n')
+cat <<EOF
+scontrol:
+$control
+EOF
+
 if [[ "$exit_status" -eq 0 ]]; then
   state=$( echo "$control" | grep '^JobState=' | cut -d= -f2)
   reason=$(echo "$control" | grep '^Reason='   | cut -d= -f2)
   if [[ "$reason" == "None" ]]; then
     reason=""
   fi
 
-  start_time=$(echo "$control" | grep '^StartTime=' | cut -d= -f2)
+  if [[ "$(echo "$control" | grep "^NodeList=" | cut -d= -f2)" == "(null)" ]]; then
+    # Skip setting the start_time when there is no allocation
+    start_time=""
+  else
+    # Set the start_time if nodes where allocated to the job
+    start_time=$(echo "$control" | grep '^StartTime=' | cut -d= -f2)
+  fi
+
   end_time=$(  echo "$control" | grep '^EndTime='   | cut -d= -f2)
-elif [[ "$control" == "slurm_load_jobs error: Invalid job id specified" ]]; then
+elif [[ "$raw_control" == "slurm_load_jobs error: Invalid job id specified" ]]; then
   # Fallback to sacct if scontrol does not recognise the ID
-  acct=$(sacct --noheader --parsable --jobs "$1" --format State,Reason,START,END  | head -n1)
+  raw_acct=$(sacct --noheader --parsable --jobs "$1" --format State,Reason,START,END,AllocTRES)
   exit_status="$?"
+  acct=$(echo "$raw_acct" | head -n1)
+  cat <<EOF
+
+sacct:
+$raw_acct
+EOF
```

Comment:
Is this debugging code that should be removed?

Reply:
It is debugging code, but I decided to leave it in so it becomes part of the log file when `log_level` is set to `DEBUG`.

The JSON payload is limited to the last line, so it doesn't make a difference. However I can redirect it to STDERR as an alternative.

Reply:
Please redirect it to STDERR.

Reply:
ping

Reply:
https://github.com/openflighthpc/flight-job/pull/14/commits/8dbac91b8f831737f4e38e773d163dac7a27bbc5

Reply:
:facepalm:

Reply:
https://github.com/openflighthpc/flight-job/pull/14/commits/bd68dd00f25388794df831dc5bc8e97ca9087545